api/models.py

```diff
@@ -620,6 +620,65 @@ class FirstStage(models.Model):
     launcher = models.ForeignKey(Launcher, related_name='firststage', on_delete=models.CASCADE)
     rocket = models.ForeignKey(Rocket, related_name='firststage', on_delete=models.CASCADE)
 
+    @property
+    def previous_flight_id(self):
+        cache_key = "%s-%s" % (self.id, "previous_flight_id")
+        res = cache.get(cache_key)
+        if res:
+            return res
+
+        launch_net = Launch.objects.get(id=self.rocket.launch.id).net
+        last_launch = Launch.objects.filter(rocket__firststage__launcher__id=self.launcher.id).filter(net__lt=launch_net).order_by('-net').first()
+
+        if last_launch:
+            res = last_launch.id
+
+        cache.set(cache_key, res, CACHE_TIMEOUT_ONE_DAY)
+        return res
+
+    @property
+    def previous_flight_name(self):
```

Comment:
After reflection - I think the @property should be def previous_flight(self) and just return the launch object. That way the API or webside can use the full object as needed.

Reply:
Okay. You reckon replace name + ID with the full object then?

Reply:
Correct - full object - then have as serializer that's just name and ID for the API.

Reply:
Okay. Not back until later tonight but will do it.